multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/PropertiesResolver.java

```diff
@@ -102,14 +102,24 @@ private boolean isSimpleReference(String value, List<Reference> references) {
     }
 
     protected Object resolveReferenceInContext(String key, Reference reference) {
+        return resolveReferenceInContext(key, reference, false);
+    }
+
+    protected Object resolveReferenceInContext(String key, Reference reference, boolean shouldBackupContext) {
         boolean resolutionContextWasCreated = false;
+        HashSet<String> contextKeysBackup = null;
```

Comment:
Isn't it better to use `Collections.emptySet()` for collections instead of null.